src/Tinfoil/AEAD/AESGCM/Data.hs

```diff
@@ -43,6 +45,11 @@ data InvocationField =
     InvocationField !RandomField !InvocationCount
   deriving (Eq, Show)
 
+packInvocationField :: InvocationField -> Word64
+packInvocationField (InvocationField (RandomField rf) (InvocationCount ic)) =
+  -- ic is guaranteed to be non-negative and less than 2^32.
+  ((fromIntegral rf :: Word64) `shiftL` 32) .|. (fromIntegral ic :: Word64)
+
```

Comment:
I notice that:
```
newtype InvocationCount =
  InvocationCount {
    unInvocationCount :: Integer
  } deriving (Eq, Show, Ord, Num)
```
Is there any reason that isn't a `Word64` instead of `Integer`. That would avoid the need for `fromIntegral`.